mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,583 @@
+            {
+                for (_, vout) in transaction.output.iter().enumerate() {
+                    if scripts.contains(&vout.script_pubkey) {
+                        let script_hash = hex::encode(electrum_script_hash(vout.script_pubkey.as_ref()));
+                        let history = ok_or_retry_after_sleep!(
+                            client.scripthash_get_history(&script_hash).compat().await,
+                            TRY_LOOP_INTERVAL
+                        );
+                        for item in history {
+                            let rpc_txid = H256Json::from(txid.as_hash().into_inner()).reversed();
+                            if item.tx_hash == rpc_txid && item.height > 0 {
+                                let height = item.height as u64;
+                                let header =
+                                    ok_or_retry_after_sleep!(get_block_header(client, height).await, TRY_LOOP_INTERVAL);
+                                let index = ok_or_retry_after_sleep!(
+                                    client
+                                        .blockchain_transaction_get_merkle(rpc_txid, height)
+                                        .compat()
+                                        .await,
+                                    TRY_LOOP_INTERVAL
+                                )
+                                .pos;
+                                let confirmed_transaction_info = ConfirmedTransactionInfo::new(
+                                    txid,
+                                    header,
+                                    index,
+                                    transaction.clone(),
+                                    height as u32,
+                                );
+                                confirmed_registered_txs.push(confirmed_transaction_info);
+                                self.registered_txs.lock().remove(&txid);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        confirmed_registered_txs
+    }
+
+    async fn append_spent_registered_output_txs(
+        &self,
+        transactions_to_confirm: &mut Vec<ConfirmedTransactionInfo>,
+        client: &ElectrumClient,
+    ) {
+        let mut outputs_to_remove = Vec::new();
+        let registered_outputs = self.registered_outputs.lock().clone();
+        for output in registered_outputs {
+            if let Some(tx_info) = ok_or_continue_after_sleep!(
+                find_watched_output_spend_with_header(client, &output).await,
+                TRY_LOOP_INTERVAL
+            ) {
+                if !transactions_to_confirm
+                    .iter()
+                    .any(|info| info.txid == tx_info.tx.txid())
+                {
+                    let rpc_txid = H256Json::from(tx_info.tx.txid().as_hash().into_inner()).reversed();
+                    let index = ok_or_retry_after_sleep!(
```

Comment:
It seems that it's worth creating `RetryableFuture` instead of using macro. So it will look like
```rust
client
    .blockchain_transaction_get_merkle(rpc_txid, tx_info.block_height)
    .compat()
    .retry_every(TRY_LOOP_INTERVAL)
    .await;
```

Reply:
I think this can be moved to the next PR as I mentioned here https://github.com/KomodoPlatform/atomicDEX-API/pull/1240#discussion_r854299918 since this might be part of the code that will be changed. If `ok_or_retry_after_sleep` is to be used I will look into creating `RetryableFuture` instead for sure :)

Reply:
Ok, I added it to the issue checklist https://github.com/KomodoPlatform/atomicDEX-API/issues/1045. Please also update it if something is done or not actual anymore, etc.

Reply:
> Please also update it if something is done or not actual anymore, etc.

Sure, I always do that, especially after a related PR merge to dev. For unresolved comments like this, I usually just reference them in the next PR that resolves them as I did here https://github.com/KomodoPlatform/atomicDEX-API/pull/1240#issue-1188101764.